server.js

```diff
@@ -77,9 +83,11 @@ app.post('/api/v1/waves/', (request, response) => {
 app.delete('/api/v1/sources/:id', (request, response) => {
   const { id } = request.params;
 
-  database('sources').where({SOURCE_ID: id}).del()
+  database('waves').where({SOURCE_ID: id}).del().then(() => {
+    database('sources').where({SOURCE_ID: id}).del()
```

Comment:
deleting all waves from the source so that no foreign keys are linked to it, blocking the deletion.